core/src/main/scala/com/salesforce/op/stages/impl/feature/SmartTextVectorizer.scala

```diff
@@ -81,23 +81,27 @@ class SmartTextVectorizer[T <: Text](uid: String = UID[SmartTextVectorizer[T]])(
     require(!dataset.isEmpty, "Input dataset cannot be empty")
 
     val maxCard = $(maxCardinality)
+    val minLenStdDev = $(minLengthStdDev)
     val shouldCleanText = $(cleanText)
 
     implicit val testStatsMonoid: Semigroup[TextStats] = TextStats.monoid(maxCard)
     val valueStats: Dataset[Array[TextStats]] = dataset.map(_.map(computeTextStats(_, shouldCleanText)).toArray)
     val aggregatedStats: Array[TextStats] = valueStats.reduce(_ + _)
 
-    val (isCategorical, topValues) = aggregatedStats.map { stats =>
+    val (isCategorical, isIgnorable, topValues) = aggregatedStats.map { stats =>
       val isCategorical = stats.valueCounts.size <= maxCard
+      val isIgnorable = stats.lengthStdDev <= minLenStdDev
```

Comment:
is this really all you need to identify IDs?

Reply:
This is the simplest check that worked on most of the synthetic data. The topK counts from a count-min sketch worked about as well, but it was harder to construct a threshold that worked across different dataset sizes and required more changes to the TextStats class. We could add that in the future, but I wanted to try out the trivial one on real data first.

The best performing criteria was a goodness of fit threshold on how well the text lengths follow an MLE Poisson fit. I was going to add that one in another PR, but could put it in this PR too. It's simple enough that it shouldn't need any curve fitting libraries.